app/services/clients/products_client.js

```diff
@@ -96,9 +96,29 @@ function getProductsByGatewayAccountId (gatewayAccountId) {
   }).then(products => products.map(product => new Product(product)))
 }
 
+/**
+ * @param {String} productExternalId: the external id of the product you wish to retrieve
+ * @returns {Promise<Product>}
+ */
+function updateProductServiceName (productExternalId, serviceName) {
```

Comment:
This looks wrong. its based on gateway account (not per product). So you need to pass the gateway account id and the new service name. The sync code will look identical to the connector name sync call.

Reply:
damn you're right.

Reply:
he is always right... :(

Reply:
sorted :-)